internal/cmd/gocdk/internal/static/_assets/demo/demo_docstore.go

```diff
@@ -0,0 +1,257 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"html/template"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"gocloud.dev/docstore"
+	_ "gocloud.dev/docstore/dynamodocstore"
+	_ "gocloud.dev/docstore/firedocstore"
+	_ "gocloud.dev/docstore/memdocstore"
+	_ "gocloud.dev/docstore/mongodocstore"
+)
+
+// TODO(rvangent): This file is user-visible, add many comments explaining
+// how it works.
+
+func init() {
+	http.HandleFunc("/demo/docstore/", docstoreBaseHandler)
+	http.HandleFunc("/demo/docstore/list", docstoreListHandler)
+	http.HandleFunc("/demo/docstore/edit", docstoreEditHandler)
+}
+
+var collectionURL string
+var collection *docstore.Collection
+var collectionErr error
+
+func init() {
+	collectionURL = os.Getenv("DOCSTORE_COLLECTION_URL")
+	if collectionURL == "" {
+		collectionURL = "mem://mycollection/Key"
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	collection, collectionErr = docstore.OpenCollection(ctx, collectionURL)
+}
+
+type MyDocument struct {
+	Key   string
+	Value string
+	// TODO(rvangent): Rename this field to demonstrate setting it once
+	// https://github.com/google/go-cloud/issues/2413 is fixed.
+	DocstoreRevision interface{}
+}
+
+// TODO(rvangent): This is pretty raw HTML. Should we have a common style sheet/etc. for demos?
```

Comment:
common style sheet for demos sounds like a good idea to me!

Reply:
Ack. I'll file an issue.